napari/layers/utils/text.py

```diff
@@ -1,242 +1,96 @@
 import warnings
-from typing import Tuple, Union
+from typing import Optional, Tuple, Union
 
 import numpy as np
+from pydantic import PositiveInt, validator
 
 from ...utils.colormaps.standardize_color import transform_color
-from ...utils.events import EmitterGroup, Event
+from ...utils.events import EventedModel
 from ...utils.translations import trans
 from ..base._base_constants import Blending
 from ._text_constants import Anchor, TextMode
 from ._text_utils import format_text_properties, get_text_anchors
 
 
-class TextManager:
+class TextManager(EventedModel):
     """Manages properties related to text displayed in conjunction with the layer.
 
     Parameters
     ----------
     text : array or str
-        the strings to be displayed
-    rotation : float
-        Angle of the text elements around the anchor point. Default value is 0.
-    anchor : str
-        The location of the text origin relative to the bounding box.
-        Should be 'center', 'upper_left', 'upper_right', 'lower_left', or 'lower_right'
-    translation : np.ndarray
-        Offset from the anchor point.
-    color : array or str
-        Font color for the text
-    size : float
-        Font size of the text. Default value is 12.
-    blending : str
-        The blending mode that determines how RGB and
-        alpha values of the layer visual get mixed. Allowed values are
-        {'opaque', 'translucent', and 'additive'}. Note that 'opaque` blending
-        is not recommended, as colors the bounding box surrounding the text.
-        The default value is 'translucent'
-    visible : bool
-        Set to true of the text should be displayed.
+        The strings to be displayed, or a format string that will be filled out
+        n_text times using data in properties.
+    n_text : int
+        The number of the strings to be displayed. This may be different to the
+        if text is a format string, or if text should be repeated.
+    properties: dict
+        Stores properties data that will be used to generate strings when text
+        is a format a string.
 
     Attributes
     ----------
-    text : array
-        the strings to be displayed
-    rotation : float
-        Angle of the text elements around the anchor point. Default value is 0.
-    anchor : str
+    values : np.ndarray
+        The text values to be displayed.
+    visible : bool
+        True if the text should be displayed, false otherwise.
+    size : float
+        Font size of the text, which must be positive. Default value is 12.
+    color : array
+        Font color for the text as an [R, G, B, A] array.
+    blending : Blending
+        The blending mode that determines how RGB and alpha values of the layer
+        visual get mixed. Allowed values are {'opaque', 'translucent', and 'additive'}.
+        Note that 'opaque` blending is not recommended, as it colors the bounding box
```

Comment:
I think blending is not allowed looking at code below? Just tried it and got

```python
viewer.layers[1].text.blending = 'opaque'
/Users/nsofroniew/GitHub/napari/napari/layers/utils/text.py:227: RuntimeWarning: opaque blending mode is not allowed for text. setting to translucent.!
```

Do you think we should allow but leave advice against, or continue to forbid?

Reply:
My instinct is to keep the behavior with the warning (i.e. forbid opaque and coerce it to translucent) and update this docstring, especially this behavior was part of the [original TextManager PR](https://github.com/napari/napari/pull/1374). Would be useful to get confirmation from @kevinyamauchi

Reply:
I am also in favor of keeping the "forbid opaque and coerce to translucent" approach. The reason we originally disallowed it is that the opaque blending is pretty strange (https://github.com/napari/napari/pull/600#issuecomment-554142225)